alacritty/src/display/mod.rs

```diff
@@ -980,74 +1020,203 @@ impl Display {
     }
 
     /// Format search regex to account for the cursor and fullwidth characters.
-    fn format_search(size_info: &SizeInfo, search_regex: &str, search_label: &str) -> String {
-        // Add spacers for wide chars.
-        let mut formatted_regex = String::with_capacity(search_regex.len());
-        for c in search_regex.chars() {
-            formatted_regex.push(c);
-            if c.width() == Some(2) {
-                formatted_regex.push(' ');
+    fn format_search(search_regex: &str, search_label: &str, max_width: usize) -> String {
+        let label_len = search_label.len();
+
+        // If the label doesn't fit into screen return.
+        if label_len > max_width {
+            return search_label[..max_width].to_owned();
+        }
+
+        // The search string consists of `search_label` + `search_regex` + `cursor`.
+        let mut bar_text = String::from(search_label);
+        bar_text.extend(
+            Self::format_visible_text(
+                iter::once('_').chain(search_regex.chars().rev()),
```

Comment:
Wait why do we pass the text in reverse here, without any docs to explain it?

Passing it in reverse just to then iterate over the allocated string and reversing it again seems highly inefficient.

Reply:
We kind of have to, since we should know truncate from the start and we have to know the length. Also we can't return iterator from that method, since we mutate the string, e.g. shorten it Though, I think we somehow look ahead it may work?

I'll try to make it iterator so we can chain `rev`, but I'm not sure how to approach it without, since we shouldn't yeild element we replace later on for example, so we need to know in advance that element is safe to yield.